bin/aws-cli-mfa-fish

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/env fish
+
+# you MUST source this file to get the exports in your shell!
+if test "$_" = source || test -n "$IS_SOURCED"
+    set SOURCED 1
+else
+    set SOURCED 0
+end
+
+set -l REQUIREMENTS python3 aws jq
+
+for requirement in $REQUIREMENTS
+    if not command -v $requirement &> /dev/null
+        echo "$requirement could not be found!"
+        exit 0
```

Comment:
This is different from the others: they only quit at this point if it was sourced -- the reason for this is to support getting usage instructions without sourcing and without installing all requirements first

Reply:
I'm not a bash expert, but I'm a little confused here.  

Using this little test script, if I understood your intent correctly, sourcing it should exit out in the `if`, but calling it directly should print out that python couldn't be found, and then print Yoooo?  That wasn't the case in my testing, but I'm probably missing something here.

```bash
#!/usr/bin/env bash

([[ -n $ZSH_EVAL_CONTEXT && $ZSH_EVAL_CONTEXT =~ :file$ ]] ||
 [[ -n $KSH_VERSION && $(cd "$(dirname -- "$0")" &&
    printf '%s' "${PWD%/}/")$(basename -- "$0") != "${.sh.file}" ]] ||
 [[ -n $BASH_VERSION ]] && (return 0 2>/dev/null)) &&
    SOURCED=1 || SOURCED=0

if ! command -v python1 >/dev/null 2>&1; then
    echo "python3 could not be found!"
    [[ $SOURCED -eq 1 ]] && return 0 || exit 0
fi

echo "Yoooo"
```

Reply:
Ah, turns out I misremembered: it is because `return` means something different in a bash script, depending on if it was sourced or not -- so maybe this does not apply to fish.

Test script I used:
```
#!/usr/bin/env bash

# https://stackoverflow.com/questions/2683279/how-to-detect-if-a-script-is-bein$
([[ -n $ZSH_EVAL_CONTEXT && $ZSH_EVAL_CONTEXT =~ :file$ ]] ||
 [[ -n $KSH_VERSION && $(cd "$(dirname -- "$0")" &&
    printf '%s' "${PWD%/}/")$(basename -- "$0") != "${.sh.file}" ]] ||
 [[ -n $BASH_VERSION ]] && (return 0 2>/dev/null)) &&
    SOURCED=1 || SOURCED=0

echo $SOURCED

if ! command -v aws1 >/dev/null 2>&1; then
    echo "aws could not be found!"
    [[ $SOURCED -eq 1 ]] && return 0 || exit 0
#    return 0
fi

echo "completed"
```

My experiments with it, first with the actual SOURCED check && return || exit, then with just return:
```
bash-3.2$ . test.sh
1
aws could not be found!
bash-3.2$ ./test.sh
0
aws could not be found!
bash-3.2$ nano test.sh # swap out commented line to enable just return 0
bash-3.2$ ./test.sh
0
aws could not be found!
./test.sh: line 15: return: can only `return' from a function or sourced script
completed
bash-3.2$ . test.sh
1
aws could not be found!
```

Reply:
Indeed, adapting my example for fish shows that fish does not have this particular bash quirk:

Script (hardcoded SOURCED):
```
#!/usr/bin/env fish

set SOURCED 0

echo $SOURCED

if not command -v aws1 &>/dev/null
    echo "aws could not be found!"
#    [[ $SOURCED -eq 1 ]] && return 0 || exit 0
    exit 0
end

echo "completed"
```

Output:
```
chris.rogus@BZUSC02YT0FALVCH ~> . test.sh
1
aws could not be found!
chris.rogus@BZUSC02YT0FALVCH ~> ./test.sh
1
aws could not be found!
chris.rogus@BZUSC02YT0FALVCH ~> nano test.sh # change SOURCED to 0
chris.rogus@BZUSC02YT0FALVCH ~> ./test.sh
0
aws could not be found!
chris.rogus@BZUSC02YT0FALVCH ~> . test.sh
0
aws could not be found!
```

Reply:
So, this is not a problem in fish -- resolving